lib/NewPost.test.js

```diff
@@ -0,0 +1,151 @@
+const NewPost = require('./NewPost');
+const mongoose = require('mongoose');
+const request = require('supertest');
+const app = require('./app');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+
+const mongodb = new MongoMemoryServer();
+
+describe('routes', () => {
+    beforeAll(() => {
+        return mongodb.getUri()
+            .then(uri => mongoose.connect(uri, {
+                useUnifiedTopology: true,
+                useNewUrlParser: true
+            }));
+    });
+    
+    beforeEach(() => {
+        return mongoose.connection.dropDatabase();
+    });
+
+    afterAll(() => {
+        return mongoose.connection.close()
+            .then(() => mongodb.stop());
+    });
+
+    it('creates a new post', () => {
+        return request(app)
+            .post('/new-post')
+            .send({
+                url: 'http://jojojo.com',
+                description: 'woah, how cool!'
+            })
+            .then(res => {
+                expect(res.body).toEqual({
+                    _id: expect.anything(),
+                    url: 'http://jojojo.com',
+                    description: 'woah, how cool!',
+                    likes: 0,
+                    __v: 0
+                });
+            });
+    });
+
+    it('returns a list of every post', () => {
+        NewPost.create(
```

Comment:
you need to await this. it is a promise. if you dont await, the code will continue to execute before the NewPost is created